cabal/CabalDependenciesMacros.hs

```diff
@@ -30,6 +37,26 @@ import System.Environment (getArgs)
 readGenericPackageDescription = readPackageDescription
 #endif
 
+#if MIN_VERSION_Cabal(3,0,0)
+generate = Macros.generateCabalMacrosHeader
+#else
+generate = Macros.generate
+#endif
+
+#if MIN_VERSION_Cabal(3,0,0)
+externalPackageDeps :: LocalBuildInfo.LocalBuildInfo -> [(UnitId, MungedPackageId)]
+externalPackageDeps lbi =
+    -- TODO:  what about non-buildable components?
+    nub [ (ipkgid, pkgid)
+        | clbi            <- Graph.toList (LocalBuildInfo.componentGraph lbi)
```

Comment:
This `Graph.toList` call is duplicated below. Please move it out to a local binding, as it will improve readability.

In my own version, I've used simply the `componentNameMap`, rather than going through the graph (we don't care about the structure here), but it's the same so good as it is.